script/lib/omaha.py

```diff
@@ -0,0 +1,149 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+import base64
+import cryptography
+import logging
+import os
+
+from lib.config import get_brave_version, get_chrome_version, get_raw_version
+from lib.connect import post, get
+from lib.helpers import release_channel
+from lib.util import omaha_channel
+
+# Event IDs never change
+event_id = {
+    'preinstall': 0,
+    'install': 1,
+    'postinstall': 2,
+    'update': 3
+}
+
+# Platform IDs never change
+platform_id = {
+    'win': 1,
+    'mac': 2
+}
+
+
+def get_channel_id(channel, host, headers, logging):
+    channel_json = get_channel_ids_from_omaha_server(host, headers, logging)
+    id = next(item["id"] for item in channel_json if item["name"] == channel)
+    return id
+
+
+def get_channel_ids_from_omaha_server(host, headers, logging):
+    """
+    Channel IDs can change between Omaha servers and over time
+    may even change on the same Omaha server. Get all IDs from
+    the server versus hardcoding in an enum like we do above
+    with Event IDs or Platform IDs.
+    """
+    url = 'http://' + host + '/api/channel'
+    response = get(url, headers)
+    if response.status_code != 200:
+        logging.error("ERROR: Cannot GET /api/channel from Omaha host {}! "
+                      "response.status_code : {}".format(host, response.status_code))
+        exit(1)
+    return response.json()
+
+
+def get_event_id(event):
+    return event_id[event]
+
+
+def get_platform_id(platform):
+    if 'win' in platform:
+        platform = 'win'
+    if 'darwin' in platform:
+        platform = 'mac'
+    return platform_id[platform]
+
+
+def get_base64_authorization(omahaid, omahapw):
+    """
+    Returns a base64 encoded string created from the Omaha ID and PW
+    """
+
+    concatstr = omahaid + ':' + omahapw
+    return base64.b64encode(concatstr.encode())
+
+
+# To-Do: Add functions to create apps
+def get_appguid(channel):
+    if channel in 'dev':
+        return '{CB2150F2-595F-4633-891A-E39720CE0531}'
+    elif channel in 'beta':
+        return '{CB2150F2-595F-4633-891A-E39720CE0531}'
+    elif channel in 'release' or channel in 'stable':
+        return '{AFE6A462-C574-4B8A-AF43-4CC60DF4563B}'
```

Comment:
is it alright to expose this publicly?

Reply:
Yes, these are already public in this repo(https://github.com/brave/brave-core/blob/master/build/config.gni#L51-L61), as well as sparkle/omaha communication if you were to look at the traffic between the omaha server and your client machine.